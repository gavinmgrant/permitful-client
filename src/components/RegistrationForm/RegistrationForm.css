.registration-form {
    margin: 10px 50px 10px 50px;
    text-align: left;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
}

.registration-container {
    margin: 30px;
}

.registration-button {
    margin-bottom: 30px;
}

@media screen and (min-width: 600px) {
    .registration-container {
        margin: 50px;
        padding: 30px;
    }
}

@media screen and (min-width: 1024px) {
    .registration-container {
        margin: 100px;
        padding: 30px;
    }
}