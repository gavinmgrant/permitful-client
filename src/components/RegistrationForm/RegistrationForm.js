import React, { Component } from 'react';
import { Button, Input, Required } from '../../utils/Utils';
import AuthApiService from '../../services/auth-api-service';
import { Link } from 'react-router-dom';
import './RegistrationForm.css';

export default class RegistrationForm extends Component {  
    state = { 
        error: null,
        onRegistrationSuccess: false
    };

    // posts user credentials to the server once user submits
    handleSubmit = ev => {
        ev.preventDefault()
        const { user_name, password } = ev.target
    
        this.setState({ error: null })
        AuthApiService.postUser({
          user_name: user_name.value,
          password: password.value,
        })
        .then(user => {
            user_name.value = ''
            password.value = ''
            this.setState({ onRegistrationSuccess: true })
        })
        .catch(res => {
            this.setState({ error: res.error })
        })
    };

    render() {
        const { error } = this.state
        return (
            <section className="registration-container">
                <h2>Register</h2>
                <p>Registered users can select their favorite permits and quickly retrieve their details later.</p>
                <p>Save time and register below.</p>
                <p>Do you want to just demo being a user?{' '}
                    <span>
                        <Link to='/login'>Login here with demo credentials</Link>
                    </span>.
                </p>
                <p className="demo">Your password must contain 1 upper case, lower case, number and special character. It must also be between 8 and 72 characters.</p>
                {this.state.onRegistrationSuccess ? <p className='success'>Success! We have saved your credentials. You can now log in to save favorites.</p> : ''}
                <form onSubmit={this.handleSubmit}>
                    <section className="registration-form">
                        <div role='alert'>
                            {error && <p className='error'>{error}</p>}
                        </div>
                        <div className='input-typ'>
                            <label htmlFor="username">
                                Username <Required />
                            </label>
                            <Input
                                name='user_name'
                                type='text'
                                required
                                id='RegistrationForm__user_name'>
                            </Input>
                        </div>
                        <div className='input-typ'>
                            <label htmlFor="password">
                                Password <Required />
                            </label>
                            <Input
                                name='password'
                                type='password'
                                required
                                id='RegistrationForm__password'>
                            </Input>
                        </div> 
                    </section>
                    <Button type='submit'>
                        Register
                    </Button>
                </form>
            </section>
        )
    }
};