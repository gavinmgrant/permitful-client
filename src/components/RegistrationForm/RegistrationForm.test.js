import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RegistrationForm from './RegistrationForm';

describe(`RegistrationForm component`, () => {
    it('renders a registration form by default', () => {
      const wrapper = shallow(<RegistrationForm />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
});