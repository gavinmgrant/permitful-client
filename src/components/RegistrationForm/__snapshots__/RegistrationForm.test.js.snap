// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders a registration form by default 1`] = `
<section
  className="registration-container"
>
  <h2>
    Register
  </h2>
  <p>
    Registered users can select their favorite permits and quickly retrieve their details later. Save time and register below.
  </p>
  <form
    onSubmit={[Function]}
  >
    <section
      className="registration-form"
    >
      <div
        role="alert"
      />
      <div>
        <label
          htmlFor="username"
        >
          Username 
          <Required />
        </label>
        <Input
          id="RegistrationForm__user_name"
          name="user_name"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="password"
        >
          Password 
          <Required />
        </label>
        <Input
          id="RegistrationForm__password"
          name="password"
          required={true}
          type="password"
        />
      </div>
    </section>
    <Button
      type="submit"
    >
      Register
    </Button>
  </form>
</section>
`;
